app/src/main/java/com/simply/birthdayapp/auth/signIn/presentation/SignInScreen.kt

```diff
@@ -0,0 +1,104 @@
+package com.simply.birthdayapp.auth.signIn.presentation
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.simply.birthdayapp.commonpresentation.components.button.AuthedButton
+import com.simply.birthdayapp.commonpresentation.components.textfield.InputTextField
+import com.simply.birthdayapp.commonpresentation.theme.AuthTitleTextStyle
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun SignInScreen(
+    modifier: Modifier = Modifier,
+    viewModel: SignInViewModel = koinViewModel(),
+    navigateToMain: () -> Unit,
+    navigateToLanding: () -> Unit,
+) {
+    SignInComposable(
+        modifier = modifier,
+        navigateToMain = navigateToMain,
+        navigateToLanding = navigateToLanding,
+        saveLoggedInState = viewModel::setSignedIn
+    )
+}
+
+@Composable
+fun SignInComposable(
+    modifier: Modifier = Modifier,
+    navigateToMain: () -> Unit,
+    navigateToLanding: () -> Unit,
+    saveLoggedInState: (Boolean) -> Unit = {},
+) {
+
+    val emailText = remember { mutableStateOf("") }
+    val emailError = remember { mutableStateOf("") }
+
+    val passwordText = remember { mutableStateOf("") }
+    val passwordError = remember { mutableStateOf("") }
+
+
+    BackHandler {
```

Comment:
is we need BackHandler? 
i think bu default we has abel this functionality

Reply:
We definitely do. The case here is that when user has already logged in, the program sets SignInScreen as start destination, and when in this case we try to go back, it just simply cannot go back to the LandingScreen as it will not be found in our backStack. I hope I have covered it well.